src/protocols/mod.rs

```diff
@@ -13,8 +14,169 @@
 
     @license GPL-3.0+ <https://github.com/KZen-networks/multi-party-ed25519/blob/master/LICENSE>
 */
+use curv::arithmetic::Converter;
+use curv::cryptographic_primitives::proofs::ProofError;
+use curv::elliptic::curves::{Ed25519, Point, Scalar};
+use curv::BigInt;
+use rand::{thread_rng, Rng};
+use sha2::{Digest, Sha512};
+
 // simple ed25519 based on rfc8032
 // reference implementation: https://ed25519.cr.yp.to/python/ed25519.py
 pub mod aggsig;
 pub mod multisig;
 pub mod thresholdsig;
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct ExpendedPrivateKey {
+    pub prefix: Scalar<Ed25519>,
+    private_key: Scalar<Ed25519>,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct ExpendedKeyPair {
+    pub public_key: Point<Ed25519>,
+    expended_private_key: ExpendedPrivateKey,
+}
+
+impl ExpendedKeyPair {
+    pub fn create() -> ExpendedKeyPair {
+        let secret = thread_rng().gen();
+        Self::create_from_private_key(secret)
+    }
+
+    pub fn create_from_private_key(secret: [u8; 32]) -> ExpendedKeyPair {
```

Comment:
I suggest renaming this function to "create_from_seed" or "create_from_secret" because the private key is derived from the secret inside this function and is returned as a field of KeyPair in the output.